app/templates/ui/app/search/search.controller.js

```diff
@@ -3,21 +3,54 @@
   'use strict';
 
   angular.module('app.search')
+    .factory('SearchModel', SearchModel)
     .controller('SearchCtrl', SearchCtrl);
 
-  SearchCtrl.$inject = ['$scope', '$location', 'userService', 'MLSearchFactory'];
+  SearchModel.$inject = ['MLSearchFactory'];
+  SearchCtrl.$inject = ['$scope', '$location', 'userService', 'SearchModel'];
 
   // inherit from MLSearchController
   var superCtrl = MLSearchController.prototype;
   SearchCtrl.prototype = Object.create(superCtrl);
 
-  function SearchCtrl($scope, $location, userService, searchFactory) {
-    var ctrl = this;
+  function SearchModel(searchFactory) {
```

Comment:
Do you think it is possible to create a more generic factory or service that keeps a store of models by name. In the ctrl you would do something like `var searchModel = modelService.get('search');`. The service returns an empty model initially, but because it is a object reference it will automatically preserve any update the ctrl makes to it. If it could work that way at least..